src/service_daemon.rs

```diff
@@ -1403,15 +1440,17 @@ impl Zeroconf {
             }
         }
 
-        // Send the query on one interface per subnet.
-        let mut subnet_set: HashSet<u128> = HashSet::new();
+        // Send the query on one interface per ip version.
+        let mut multicast_notification_trackers = HashSet::new();
         for (_, intf_sock) in self.intf_socks.iter() {
-            if !intf_sock.intf.is_link_local() {
-                let subnet = ifaddr_subnet(&intf_sock.intf.addr);
-                if subnet_set.contains(&subnet) {
-                    continue; // no need to send query the same subnet again.
+            if intf_sock.as_multicast_notification_tracker().is_some() {
```

Comment:
same here for using `if let Some(tracker)`

Reply:
done